netmiko/fortinet/fortinet_ssh.py

```diff
@@ -1,104 +1,298 @@
+		where cmd_verify is automatically disabled.
+		"""
+		return super().send_config_set(
+			config_commands=config_commands,
+			exit_config_mode=exit_config_mode,
+			read_timeout=read_timeout,
+			delay_factor=delay_factor,
+			max_loops=max_loops,
+			strip_prompt=strip_prompt,
+			strip_command=strip_command,
+			config_mode_command=config_mode_command,
+			cmd_verify=cmd_verify,
+			enter_config_mode=enter_config_mode,
+			error_pattern=error_pattern,
+			terminator=terminator,
+			bypass_commands=bypass_commands
+		)
+
+	def enable(self, cmd="", pattern="", re_flags=re.IGNORECASE):
+		"""Not Supported for this platform"""
+		pass
+
+	def exit_enable_mode(self, exit_command: str = "") -> str:
+		"""Not Supported for this platform"""
+		pass
+
+	def set_terminal_width(self):
+		"""Not Supported for this platform"""
+		pass
+
+	def check_config_mode(
+		self, 
+		check_string: str = r"(\) #|\) \$)", 
+		pattern: str = r"", 
+		force_regex: bool = True
+		) -> bool:
+		"""Checks if the device is in configuration mode or not.
+
+		:param check_string: Identification of configuration mode from the device
+		:type check_string: str
+
+		:param pattern: Pattern to terminate reading of channel
+		:type pattern: str
+		"""
+		return super().check_config_mode(
+			check_string=check_string, pattern=pattern, force_regex=force_regex
+		)
+
+	def exit_config_mode(self, exit_config="end", pattern=r"(#|\$)"):
+		"""Exit from configuration mode.
+
+		:param exit_config: Command to exit configuration mode
+		:type exit_config: str
+
+		:param pattern: Pattern to terminate reading of channel
+		:type pattern: str
+		"""
+		return super().exit_config_mode(
+			exit_config=exit_config, pattern=pattern
+		)
+
+	def disable_paging(
+		self,
+		command: str = "",
+		delay_factor: Optional[float] = None,
+		cmd_verify: bool = True,
+		pattern: Optional[str] = None,
+		) -> str:
+		"""Disable paging is only available with specific roles so it may fail.
+
+		:param command: Device command to disable pagination of output
+
+		:param delay_factor: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+		"""
+
+		disable_paging_commands = [
+			"config global",
+			"config system console",
+			"set output standard",
+			"end"
+			]
+		output = self.send_config_set(
+			config_commands = disable_paging_commands,
+			exit_config_mode = True,
+			read_timeout = None,
+			delay_factor = 1.0,
+			max_loops = 150,
+			enter_config_mode = False,
+		)
+		log.debug("***Window Paging Disabled***")
+		return output
+
+	def cleanup(self, command: str = "quit") -> None:
+		"""Re-enable paging globally."""
+		# Return paging state
+		enable_paging_commands = ["config global", "config system console", "set output more", "end"]
+		# Should test output is valid
+		output = self.send_config_set(
+			config_commands = enable_paging_commands,
+			exit_config_mode = True,
+			read_timeout = None,
+			delay_factor = 1.0,
+			max_loops = 150,
+			enter_config_mode = False,
```

Comment:
Same send_config_set argument issue.

Reply:
We'll probably have to do something similar like we did with Audiocodes since Fortigates don't have a traditional configuration mode.  You basically type something like "config system global" or "config system snmp user" then that takes into a hierarchy to make configs there.  There's no "config mode" that can be define as a constant.  Also, if multi-vdom mode is enabled, you first have to go into the vdom you want to change, like "config vdom" > "edit root" > then "config router static".  So it's best to leave flexibility like we did on Audiocodes.

Reply:
Okay, we can do what we did in audiocode (i.e. require a `config mode` command, and raise an error if the user doesn't provide it). So `enter_config_mode = True` would be the default.

Reply:
So for audiocode we required:

```
enter_config_mode: bool = True,
```

And then left:

```
config_mode_command: Optional[str] = None,
```

And then in the code checked that `config_mode_command` MUST be set to something or Netmiko we will raise an exception, we can do that again here...that is fine with me.

Reply:
Sounds good, I'll do a draft with the changes and submit the PR.

Reply:
See the terminator thread, posted the updated config there.